vm_method.c

```diff
@@ -1027,10 +1026,16 @@ rb_method_entry_make(VALUE klass, ID mid, VALUE defined_class, rb_method_visibil
                 break;
             }
             if (iseq) {
-                rb_compile_warning(RSTRING_PTR(rb_iseq_path(iseq)),
-                                   ISEQ_BODY(iseq)->location.first_lineno,
-                                   "previous definition of %"PRIsVALUE" was here",
-                                   rb_id2str(old_def->original_id));
+                rb_warning(
+                    "method redefined; discarding old %"PRIsVALUE", previous definition of %"PRIsVALUE" was at %s:%d",
```

Comment:
This format can't tag-jump to the previous definition.
Keep two lines with one `Warning.warn` call.